### ISSUE TITLE
Microsoft.NETCore.UniversalWindowsPlatform is missing some packages after "exclude": "compile" updates

### ISSUE THREAD

**Issue by ericstj** [dotnet/corefx#10338]
We updated all packages to hide their implementation dependencies with "exclude": "compile".  We did this because implementation dependencies change regularly and aren't seen as part of the compatibility contract of a package.  To ensure that folks don't rely on our packages for their implementation dependencies we use `"exclude": "compile"` in the packages.

This transition caused at some package to be excluded entirely from the UWP meta-package since they were only referenced transitively.  We should consider adding these back unless we specifically want to keep them out of the default reference set.  /cc @weshaggard @terrajobst 

The dropped references are:
- System.Collections.NonGeneric
- System.Collections.Specialized
- System.Threading.Overlapped
- System.Xml.XmlDocument
## Workaround

For folks that hit missing types from one of these packages after upgrading to Microsoft.NETCore.UniversalWindowsPlatform they can reference the packages directly as follows.

``` json
    "System.Collections.NonGeneric": "4.0.1",
    "System.Collections.Specialized": "4.0.1",
    "System.Threading.Overlapped": "4.0.1",
    "System.Xml.XmlDocument": "4.0.1"
```

**Comment by weshaggard** [comment]
I would simply hold off and just include them when we rev all them as part of the api work.

**Comment by johntester000010** [comment]
Hi, ericstj! It seems that workaround doesn't work for deisgn-time if it use types from described assemblies. I have created examlple that shows error on MainPage.xaml:
Exception: Cannot create an instance of "Tester". 
at System.RuntimeTypeHandle.CreateInstance
[nuget_designer.zip](https://github.com/dotnet/corefx/files/386304/nuget_designer.zip)
Is there any way to fix the issue?

**Comment by ericstj** [comment]
@johntester000010 I think you're just dealing with the designer using stale references  I downloaded your sample and loaded it and it worked file for me.

**Comment by johntester000010** [comment]
Thanks, for your answer, @ericstj !
I just found that designer doesn't show anything if solution platform is ARM or x64
Please, set active platform to x86. I hope you will see the exception message
Before update to 5.2.2 MainPage showed type name (System.Collections.ArrayList) on red backgroun

**Comment by ericstj** [comment]
Indeed I can reproduce.
```
BadImageFormatException: Could not load file or assembly 'System.Collections.NonGeneric, Version=4.0.1.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. Reference assemblies should not be loaded for execution.  They can only be loaded in the Reflection-only loader context. (Exception from HRESULT: 0x80131058)
```

The designer isn't pulling out the correct asset from the package.  It's supposed to be using the  output of `RuntimeImplementationProjectOutputGroup` which are actual runtime assemblies, but instead it seems to be using the references passed to the compiler which are reference assemblies.  

I enabled design time tracing and added some logging to `RuntimeImplementationProjectOutputGroup` and confirm that this is passing the right dll (implementation, not reference assembly): 
```
<userprofile>\.nuget\packages\System.Collections.NonGeneric\4.0.1\lib\netstandard1.3\System.Collections.NonGeneric.dll
```

@lutzroeder I think this is a designer bug.

/cc @jasonmalinowski

**Comment by danmosemsft** [comment]
@joperezr I assume this is still relevant.

**Comment by ericstj** [comment]
I believe @zamont is fixing this.

**Comment by ericstj** [comment]
@joperezr let's make sure that these libraries are all included in the ref set for latest UWP flat package.

**Comment by zamont** [comment]
Commit dotnet/core-setup@42121ee commits this fix to the dotnet/core-setup:master branch. I'm not sure if the UWP MP package gets produced off the master branch, but if it does, then this fix will be in the frequently-built prerelease packages of Microsoft.NETCore.UniversalWindowsPlatform on myget.

**Comment by joperezr** [comment]
> @joperezr let's make sure that these libraries are all included in the ref set for latest UWP flat package.

Looks like they are all there except for System.Xml.XmlDocument. I'll send up a PR to fix this.

**Comment by zamont** [comment]
Fixed by dotnet/core-setup#1649
